addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -2304,6 +2304,76 @@ var FieldRadio = FieldSelection.extend({
     },
 });
 
+
+var FieldSelectionBadge = FieldSelection.extend({
+    template: null,
+    className: 'o_field_selection_badge',
+    tagName: 'span',
+    specialData: "_fetchSpecialMany2ones",
+    events: _.extend({}, AbstractField.prototype.events, {
+        'click span.o_selection_badge': '_onBadgeClicked',
+    }),
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @override
+     */
+    _renderEdit: function () {
+        var self = this;
+        this.currentValue = this.value;
```

Comment:
I would cleanly integrate in an el-if structure with the herebelow lines.

Reply:
We support two types, selection and many2one, and special case is many2one so we keep
```this.currentValue = this.value;```` for normal case and handled many2one specifically

Reply:
That's not my point. I think clearer to use an if - else construct than something - if . It is easier to understand you have two cases.